File: Easy Challenges/0145 - Tree Generation/Solution.py
def generate_tree(base, trunk, leaves):
    for i in range(int(base / 2 + 1)):
        print(" " * int(base / 2 - i) + leaves * (i * 2 + 1))
    print(" " * int(base / 2 - 1) + trunk * 3)


def main():
    arr_input = input("Enter the base-width, trunk, and leaves (space-delimited): ").split()
    generate_tree(int(arr_input[0]), arr_input[1], arr_input[2])

if __name__ == '__main__':
    main()
    

File: Easy Challenges/0145 - Tree Generation/Solution.java
import java.util.Scanner;

public class Solution {

    public static void main(String[] args) {
        Scanner reader = new Scanner(System.in);
        System.out.print("Enter the base-width, trunk, and leaves (space-delimited): ");
        String[] arrInput = reader.nextLine().split(" ");
        generateTree(Integer.parseInt(arrInput[0]), arrInput[1].charAt(0), arrInput[2].charAt(0));
    }

    public static void generateTree(int base, char trunk, char leaves) {
        for (int i = 0; i < base / 2 + 1; i++) {
            for (int j = 0; j < base / 2 - i; j++) {
                System.out.print(" ");
            }
            for (int j = 0; j < i * 2 + 1; j++) {
                System.out.print(leaves);
            }
            System.out.println();
        }
        for (int i = 0; i < base / 2 - 1; i++) {
            System.out.print(" ");
        }
        for (int i = 0; i < 3; i++) {
            System.out.print(trunk);
        }
    }

}
